Ex1/week5_4.py

```diff
@@ -0,0 +1,98 @@
+import itertools
+import os
+import re
+from bs4 import BeautifulSoup
+
+# Mixed Tools
+
+# -- Regular version:
+
+
+def interleave(*args):
+    """
+    Take an unlimited list of arguments that can be iterate and returns a list of the elements in order
+    for example:
+    Input: 'abc', [1, 2, 3], ('!', '@', '#')
+    Output: ['a', 1, '!', 'b', 2, '@', 'c', 3, '#']
+    :param args: An iterable
+    :return: A list of elements mixed in order
+    """
+    list_of_tuples = []
+    for elem in zip(*args):
+        list_of_tuples.append(elem)
+    return list(sum(list_of_tuples, ()))
+
+
+# -- Generator version:
+
+
+def interleave_generator(*args):
+    """
+    Take an unlimited list of arguments that can be iterate and returns a list of the elements in order
+    using generator
+    for example:
+    Input: 'abc', [1, 2, 3], ('!', '@', '#')
+    Output: ['a', 1, '!', 'b', 2, '@', 'c', 3, '#']
+      :param args: An iterable
+    :return: A list of elements mixed in order
+    """
+    for elem in zip(*args):
+        yield elem
+
+
+my_list = []
+my_list += [item for element in interleave_generator('abc', [1, 2, 3], ('!', '@', '#'))
```

Comment:
What is this code? why isn't it in a function? where is https://www.freecodecamp.org/news/if-name-main-python-example/